editor.planx.uk/src/pages/FlowEditor/components/FeedbackLog/queries/updateFeedbackStatus.ts

```diff
@@ -0,0 +1,36 @@
+import { GridRowId } from "@mui/x-data-grid";
+import gql from "graphql-tag";
+import { client } from "lib/graphql";
+
+import { FeedbackStatus } from "../types";
+
+export const updateFeedbackStatus = async (
+  selectedRowIds: readonly GridRowId[],
+  status: FeedbackStatus,
+) => {
+  await client.mutate<{ updatedFeedback: { id: number }[] }>({
+    mutation: gql`
+      mutation UpdateFeedbackStatus(
+        $status: feedback_status_enum_enum
+        $ids: [Int!]
+      ) {
+        updatedFeedback: update_feedback_many(
+          updates: { where: { id: { _in: $ids } }, _set: { status: $status } }
+        ) {
+          returning {
+            id
+          }
+        }
+      }
+    `,
+    variables: {
+      ids: selectedRowIds,
+      status,
+    },
+    update: (cache) => {
+      // As we're reading from feedback_summary but pushing an update to feedback, we need to manually invalidate Apollo's cache
+      cache.evict({ fieldName: "feedback_summary" });
+      cache.gc();
```

Comment:
:+1:

Reply:
Ah neat this is a super good observation - as it's a subscription it's already automatically pulling in updates. This would work!

I think we need to have a bit of a re-think about how we handle local state in the frontend - there's increasingly more tedious edge cases like this we need to work around.